### DISCUSSION TITLE
Naming convention

### DISCUSSION THREAD

**Discussion by Teoaks** [Cyfrin/foundry-full-course-cu#254] [General]
Is there a difference between naming conventions between constants and immutable variables?
I understood in the lesson that all capital letters and a _ is used for constants, but what about immutable ones?
Thanks in advance!

**Comment by pyromaus** [comment]
```solidity
bytes32 private immutable i_gasLane;
```
i, underscore, name of variable.

**Comment by IAm0xAxe** [comment]
Constants should be named with all capital letters with underscores separating words. Examples: MAX_BLOCKS, TOKEN_NAME, TOKEN_TICKER, CONTRACT_VERSION.

According to the official Solidity 0.8.20 documentation, there are naming conventions for "constants" but not for "immutable" variables. **Therefore, I believe it is reasonable to follow the naming conventions for "constants" when naming "immutable" variables.**

**Comment by ShubhSensei** [comment]
i for immutable `i_owner`
c for constants `c_balance`
s for storage values `s_funders`

**Reply by IAm0xAxe** [reply]
c for constants `c_balance`

**Reply by ShubhSensei** [reply]
> c for constants `c_balance`

My bad, thanx for telling me 😁

**Reply by f3d0ss** [reply]
Actually for **constant** during the course Patrick follow the [Solidity style guide](https://docs.soliditylang.org/en/latest/style-guide.html#constants) using the `UPPER_CASE_WITH_UNDERSCORES`

**Reply by IAm0xAxe** [reply]
You're Welcome 😁
